src/rule_definition_tools.jl

```diff
@@ -210,25 +210,29 @@ function propagation_expr(Δs, ∂s, _conj = false)
     ∂s = map(esc, ∂s)
     n∂s = length(∂s)
 
-    # Due to bugs in Julia 1.0, we can't use `.+`  or `.*` inside expression literals.
-    ∂_mul_Δs = if _conj
-        ntuple(i->:(conj($(∂s[i])) * $(Δs[i])), n∂s)
+    init_expr = if _conj
+        :((*).(conj($(∂s[1])), $(Δs[1])))
```

Comment:
Using `ntuple` will be better here because it guarantees to fully unroll the expression. Using broadcasting could give suboptimal performance and worse compile time.

Reply:
Sorry, I don't understand, why/how should I use `ntuple` here? Previously all partials and gradients were multiplied pairwise and `ntuple` was used to construct all these expressions, but here it is just a single expression for the first partial and gradient.

Reply:
Oh, I see. I misunderstood the structure of the new code. Yeah, this is correct.